Automattic/jetpack
# Sharing Block: add webshare option

Fixes https://github.com/Automattic/jetpack/issues/35533

## Proposed changes:

Adding a simple Web Share API implementation for native tools to share the post. Behavior is similar to https://github.com/Automattic/jetpack/pull/24473


https://github.com/Automattic/jetpack/assets/60262784/7931ab17-df3b-49d8-ba34-9faee2c9940c



The suggestion that instead of hiding the button on the devices without Native share (eg chrome on laptop) we will simply Copy the URL to clipboard. We could implement a small tooltip `Added to clipboard` if decided to go this way.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up this branch
* Navigate to new Post and add sharing buttons block
* Add `Native Share` button and check the behavior
